File: python-implementation/quicksort.py
unsorted = [5, 2, 3, 4, 1]

def quick_sort(values):
    if len(values) <= 1:
        return values

    pivot = values[0]
    left = []
    right = []

    for i in range(1, len(values)):
        if values[i] < pivot:
            left.append(values[i])
        else:
            right.append(values[i])

    left = quick_sort(left)
    right = quick_sort(right)

    return left + [ pivot ] + right

print('Expect [1,2,3,4,5]:', quick_sort(unsorted))

File: js-implementation/quick-sort.js
let unsorted = [5, 2, 3, 4, 1];

const quickSort = array => {
  if (array.length <= 1) return array;

  const pivot = array[0];
  let left = [];
  let right = [];

  for (let i = 1; i < array.length; i++) {
    const element = array[i];
    if (element < pivot) {
      left.push(element);
    } else {
      right.push(element);
    }
  }

  left = quickSort(left);
  right = quickSort(right);

  return left.concat(pivot, right);
};

console.log('Expect [1,2,3,4,5]:', quickSort(unsorted));
